src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
another thing: adapter could provide two traits for client
implementations to provide the other half. one that has the request
factory and the http methods and calls send(), another one that looks at
the request and calls the appropriate http methods.

then the two sub interfaces would never really be used alone and all
clients could implement the main interface without much effort.